nibabel/metasum.py

```diff
@@ -0,0 +1,583 @@
+# emacs: -*- mode: python-mode; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the NiBabel package for the
+#   copyright and license terms.
+#
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+'''Memory efficient tracking of meta data dicts with repeating elements
+'''
+from dataclasses import dataclass
```

Comment:
This is going to fail on Python 3.6. We may need to create a stub `dataclass` that causes `DimIndex` to fail when initialized, but avoid crashing when just importing this file.

Reply:
There is a backport available for 3.6: https://pypi.org/project/dataclasses/

Otherwise I could replace this with a more verbose regular class.

Reply:
I guess let's add a conditional dependency for 3.6.